clients/unshaded/src/main/java/tachyon/client/TachyonFS.java

```diff
@@ -216,7 +223,13 @@ synchronized void accessLocalBlock(long blockId) throws IOException {
    * @throws IOException if the underlying worker RPC fails
    */
   synchronized void addCheckpoint(long fid) throws IOException {
-    mWorkerClient.addCheckpoint(fid);
+    try {
+      mWorkerClient.addCheckpoint(fid);
+    } catch (FailedToCheckpointException e) {
```

Comment:
As per my previous comment, this should just be ``TException e`` and there should only be one catch clause.

Alternatively, if we choose to list all exception cases, we should do so consistently (e.g. requestBlockLocation() below should list all of its exceptions and not just ``TException``).